src/components/select/multi-select/multi-select.component.js

```diff
@@ -0,0 +1,416 @@
+
+  const handleGlobalClick = useCallback((event) => {
+    if (!isOpen) {
+      return;
+    }
+
+    const notInContainer = containerRef.current && !containerRef.current.contains(event.target);
+
+    if (notInContainer && !isClickTriggeredBySelect.current) {
+      setTextValue('');
+      setFilterText('');
+      setHighlightedValue('');
+      setOpenState(false);
+    }
+
+    isClickTriggeredBySelect.current = false;
+  }, [isOpen]);
+
+  const mapValuesToPills = useCallback(() => {
+    const canDelete = !disabled && !readOnly;
+
+    if (selectedValue.length === 0) {
+      return '';
+    }
+
+    return selectedValue.map((singleValue, index) => {
+      const matchingOption = React.Children.toArray(children).find(option => (option.props.value === singleValue));
+
+      return (
+        <StyledSelectPillContainer key={ singleValue }>
+          <Pill
+            onDelete={ canDelete ? () => removeSelectedValue(index) : undefined }
+            title={ matchingOption.props.text }
+          >
+            { matchingOption.props.text }
+          </Pill>
+        </StyledSelectPillContainer>
+      );
+    });
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [children, disabled, readOnly, selectedValue]);
+
+  useEffect(() => {
+    const newValue = value || defaultValue;
+    const modeSwitchedMessage = 'Input elements should not switch from uncontrolled to controlled (or vice versa). '
+    + 'Decide between using a controlled or uncontrolled input element for the lifetime of the component';
+    const onChageMissingMessage = 'onChange prop required when using a controlled input element';
+
+    invariant(isControlled.current === (value !== undefined), modeSwitchedMessage);
+    invariant(!isControlled.current || (isControlled.current && onChange), onChageMissingMessage);
+    setSelectedValue((previousValue) => {
+      if (!newValue && previousValue.length === 0) {
+        return previousValue;
+      }
+
+      return newValue;
+    });
+  }, [value, defaultValue, onChange]);
+
+  // removes placeholder when a value is present
+  useEffect(() => {
+    const hasValue = value && value.length > 0;
+    const hasSelectedValue = selectedValue && selectedValue.length > 0;
+
+    if (hasValue || hasSelectedValue) {
+      setPlaceholderOverride(' ');
+    } else {
+      setPlaceholderOverride(placeholder);
+    }
+  }, [value, selectedValue, placeholder]);
+
+  useEffect(() => {
+    const clickEvent = 'click';
+
+    document.addEventListener(clickEvent, handleGlobalClick);
+
+    return function cleanup() {
+      document.removeEventListener(clickEvent, handleGlobalClick);
+    };
+  }, [handleGlobalClick]);
+
+  useEffect(() => {
+    if (!isControlled.current && onChange) {
+      onChange(createCustomEvent(selectedValue));
+    }
+
+    setRepositionTrigger((previousValue) => {
+      return !previousValue;
+    });
+  }, [createCustomEvent, onChange, selectedValue]);
+
+  function handleTextboxClick(event) {
```

Comment:
I'd stick to defining component methods using either `const method = () => {}` or `function method() {}`. Pick one and stick with it.

Reply:
i'm defining all of these methods with `function`, only these that have to rely on `useCallback` are defined using `const` here, as other solution that I could think of is:
```
function handleSomething() {
  return useCallback(() => {
  .
  .
  .
  });
}
```
which is less readable in my opinion

Reply:
I'd just define everything using arrow functions, but I leave it up to you

Reply:
when using const you have to think where you will place your method as it cannot be below anything that uses it.
In my opinion it leads to a messy code.